fido/fido.py

```diff
@@ -110,21 +139,12 @@ def finished_errorback(error):
 
     # Fetch the body once we've received the headers
     def response_callback(response):
+        finished = Deferred()
         response.deliverBody(HTTPBodyFetcher(response, finished))
+        return finished
 
     deferred.addCallback(response_callback)
-    deferred.addErrback(finished.errback)
-
-    if timeout is not None:
-        # Cancel the request if we hit the timeout
-        def cancel_timer(response):
-            if timer.active():
-                timer.cancel()
-            return response
-        timer = reactor.callLater(timeout, deferred.cancel)
-        finished.addBoth(cancel_timer)
```

Comment:
just noticed I forgot to implement the timeout. 
will look for a solution to this.